lib/foreman_rh_cloud/engine.rb

```diff
@@ -193,4 +199,8 @@ def self.register_scheduled_task(task_class, cronline)
       end
     end
   end
+
+  def self.with_local_advisor_engine?
+    SETTINGS&.[](:foreman_rh_cloud)&.[](:use_local_advisor_engine) || false
```

Comment:
You can use [dig](https://ruby-doc.org/3.4.1/Hash.html#method-i-dig) on a hash and I think it's safe to assume `SETTINGS` is a hash (so `&` is redundant):
```suggestion
    SETTINGS.dig(:foreman_rh_cloud, :use_local_advisor_engine) || false
```